app/src/lib/popup-manager.ts

```diff
@@ -0,0 +1,118 @@
+import uuid from 'uuid'
+import { Popup, PopupType } from '../models/popup'
+import { sendNonFatalException } from './helpers/non-fatal-exception'
+
+/**
+ * The limit of how many popups allowed in the stack. Working under the
+ * assumption that a user should only be dealing with a couple of popups at a
+ * time, if a user hits the limit this would indicate a problem.
+ */
+const popupStackLimit = 50
+
+/**
+ * The popup manager is to manage the stack of currently open popups.
+ */
+export class PopupManager {
+  private popupStack = new Array<Popup>()
+
+  /**
+   * Returns the last popup added to the stack.
+   */
+  public get currentPopup(): Popup | null {
+    return this.popupStack.at(-1) ?? null
+  }
+
+  /**
+   * Returns whether there are any popups in the stack.
+   */
+  public get isAPopupOpen(): boolean {
+    return this.currentPopup !== null
+  }
+
+  /**
+   * Returns an array of all popups in the stack of the provided type.
+   **/
+  public getPopupsOfType(popupType: PopupType): ReadonlyArray<Popup> {
+    return [...this.popupStack.filter(p => p.type === popupType)]
+  }
+
+  /**
+   * Returns whether there are any popups of a given type in the stack.
+   */
+  public arePopupsOfType(popupType: PopupType): boolean {
```

Comment:
Nit? (Feel free to ignore it)
```suggestion
  public areTherePopupsOfType(popupType: PopupType): boolean {
```

Reply:
I have no preference so I will change it but funnily enough I have been on a team where `are` would be preferred over `areThere`

Reply:
I just sounded odd to me. Maybe even `hasPopupsOfType` is more common for this kind of methods? Anyway, nothing to worry much about IMHO 😄